### ISSUE TITLE
Is there any doc that tells how to extract the data available from TCP?

### ISSUE THREAD

**Issue by gbernal** [TadasBaltrusaitis/OpenFace#613]
I'm running the example code for gaze and head pose, but I'm interested in the action units.
My guess that data is also being streamed but not displayed, my question is, Is there any doc that tells how to extract the data available from TCP? 

That would be greatly appreciated!
Thank you!

**Comment by gbernal** [comment]
looking a bit closer to the code, I think what I'm looking for is for the `topic_filter` availables

**Comment by TadasBaltrusaitis** [comment]
Hi, currently the only features exposed through the messaging are gaze and head pose, but not any others. You are very welcome to add them though!

It's on my list to add the functionality, but I haven't gotten round to it.

**Comment by gbernal** [comment]
I will gladly help. Do you mind pointing me to which file is doing the messaging

**Comment by TadasBaltrusaitis** [comment]
The part of the code that does messaging at the moment is in `\gui\HeadPose-live\MainWindow.xaml.cs`, it uses ZeroMQ for the messaging (you can look for parts of the code referencing it). This was built for a special application in mind and is not very generic.

Ideally the messaging should be done through a C++ interface, maybe even as part of `RecorderOpenFace`, where the target specified would be not only a file, but also a location to broadcast to. This would allow the messaging to be called from any of the executables easily and in a generic way.

**Comment by gbernal** [comment]
Great thank you for the info. I spent some time yesterday looking at the recordCSV.cpp file and I was thinking just to add the tcp portion there. 
I will look into the files you mentioned and I will try to implement in recordedOpenFace

**Comment by marc101101** [comment]
@TadasBaltrusaitis is this messaging currently only available in a certain part of the software? Because when I start $./FaceLandmarkVid -device0 nothing is published to the port 5000 (referring to your python scripts)

**Comment by TadasBaltrusaitis** [comment]
Yes, unfortunately, it is currently only part of the `HeadPoseLive` executable for Windows. It is on my TODO but I haven't gotten round to it yet.
